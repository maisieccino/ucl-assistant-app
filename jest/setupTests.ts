import '@testing-library/jest-native/extend-expect'
import '@testing-library/react-native/cleanup-after-each'
import 'react-native-gesture-handler/jestSetup'

// Silence the warning: Animated: `useNativeDriver` is not supported because
// the native animated module is missing
// https://github.com/facebook/react-native/issues/11094#issuecomment-263240420
jest.mock(`react-native/Libraries/Animated/src/NativeAnimatedHelper`)
