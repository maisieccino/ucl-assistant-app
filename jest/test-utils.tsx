import { NavigationContainer } from '@react-navigation/native'
import { render as rtlRender, RenderResult } from '@testing-library/react-native'
import React from 'react'
import { Provider } from 'react-redux'
import { Store } from 'redux'
import configureStore from 'redux-mock-store'
import thunk from "redux-thunk"
import debounce from "../lib/debounce"
import { initialState as defaultInitialState } from '../reducers'

const middlewares = [
  debounce.middleware,
  thunk,
]
const mockStore = configureStore(middlewares)

const render = (
  ui: React.ReactElement,
  {
    initialState = defaultInitialState,
    store = mockStore(initialState) as Store,
    ...renderOptions
  } = {},
): RenderResult => {
  const Wrapper = ({ children }) => (
    <Provider store={store}>
      <NavigationContainer>
        {children}
      </NavigationContainer>
    </Provider>
  )
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions })
}

export const waitForEventLoop = async (): Promise<void> => new Promise((resolve) => {
  const done = () => {
    jest.useFakeTimers()
    resolve()
  }
  jest.useRealTimers()
  setTimeout(done, 0)
})

// re-export everything
export * from '@testing-library/react-native'
// override render method
export { render }
