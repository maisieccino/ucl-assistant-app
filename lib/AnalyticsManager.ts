import * as Amplitude from 'expo-analytics-amplitude'
import Constants from 'expo-constants'

import configureStore from "../configureStore"

const { store } = configureStore

const AMPLITUDE_API_KEY = Constants.manifest.extra
  && Constants.manifest.extra.amplitude
  && Constants.manifest.extra.amplitude.apiKey
  ? Constants.manifest.extra.amplitude.apiKey
  : null

const initialise = (): Promise<void> => Amplitude.initialize(AMPLITUDE_API_KEY)

const shouldTrackAnalytics = (): boolean => (
  AMPLITUDE_API_KEY
  && !__DEV__
  && store.getState().user.settings.shouldTrackAnalytics
)

const setUserId = (userId): void => {
  if (!shouldTrackAnalytics()) { return }

  // make sure this is unique
  if (userId) {
    Amplitude.setUserId(userId)
  }
}

const setUserProperties = ({ ...userProperties }): void => {
  if (!shouldTrackAnalytics()) { return }

  Amplitude.setUserProperties({ ...userProperties })
}

const clearUserProperties = (): void => {
  Amplitude.clearUserProperties()
}

const logEvent = (eventName: string, eventProperties?): void => {
  if (!shouldTrackAnalytics()) { return }

  if (eventProperties) {
    Amplitude.logEventWithProperties(eventName, eventProperties)
  } else {
    Amplitude.logEvent(eventName)
  }
}

const logScreenView = (screenName: string): void => {
  if (!shouldTrackAnalytics()) { return }

  logEvent(`VIEW_SCREEN_${screenName}`)
}

const events = {
  NOTIFICATIONS_ENABLE: `NOTIFICATIONS_ENABLE`,
  NOTIFICATIONS_SKIP: `NOTIFICATIONS_SKIP`,
  PUSH_NOTIFICATIONS_REGISTER: `PUSH_NOTIFICATIONS_REGISTER`,
  SETTINGS_GIVE_FEEDBACK: `SETTINGS_GIVE_FEEDBACK`,
  SETTINGS_RATE_APP: `SETTINGS_RATE_APP`,
  SETTINGS_VIEW_FAQS: `SETTINGS_VIEW_FAQS`,
}

export const AnalyticsManager = {
  clearUserProperties,
  events,
  initialise,
  logEvent,
  logScreenView,
  setUserId,
  setUserProperties,
}

export default AnalyticsManager
