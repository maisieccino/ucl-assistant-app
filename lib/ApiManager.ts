import axios from "axios"
import { encode as btoa } from "base-64"

import {
  EQUIPMENT_URL,
  FREE_ROOMS_URL,
  ROOMBOOKINGS_URL,
  ROOMS_URL,
  SITES_URL,
  TIMETABLE_URL,
  WORKSPACES_URL,
} from "../constants/API"
import ErrorManager from "./ErrorManager"
import LocalisationManager from './LocalisationManager'

const roomsController = {
  getBookings: async (
    token = null,
    {
      roomid,
      siteid,
      date = LocalisationManager.getMoment().format(`YYYYMMDD`),
    },
  ) => {
    if (!roomid || !siteid) {
      throw new Error(`Must specify roomid and siteid`)
    }
    try {
      const results = await axios.get(ROOMBOOKINGS_URL, {
        headers: {
          authorization: `Bearer ${token}`,
        },
        params: {
          date,
          roomid,
          siteid,
        },
      })
      if (!results.data.content.ok) {
        throw new Error(results.data.content.error)
      }
      return results.data.content.bookings
    } catch (error) {
      ErrorManager.captureError(error)
      throw error
    }
  },
  getEmptyRooms: async (
    token = null,
    {
      startDateTime = (new Date()).toISOString(),
      endDateTime = (
        LocalisationManager
          .getMoment()
          .add(1, `hours`)
          .toISOString()
      ),
    } = {},
  ) => {
    try {
      const results = await axios.get(FREE_ROOMS_URL, {
        headers: {
          authorization: `Bearer ${token}`,
        },
        params: {
          endDateTime,
          startDateTime,
        },
      })
      if (!results.data.content.ok) {
        throw new Error(results.data.content.error)
      }
      return results.data.content.free_rooms
    } catch (error) {
      ErrorManager.captureError(error)
      throw error
    }
  },
  getEquipment: async (token = null, { roomid, siteid }) => {
    if (!roomid || !siteid) {
      throw new Error(`Must specify roomid and siteid`)
    }
    try {
      const results = await axios.get(EQUIPMENT_URL, {
        headers: {
          authorization: `Bearer ${token}`,
        },
        params: {
          roomid,
          siteid,
        },
      })
      if (!results.data.content.ok) {
        throw new Error(results.data.content.error)
      }
      return results.data.content.equipment
    } catch (error) {
      ErrorManager.captureError(error)
      throw error
    }
  },
  getSites: async (token = null) => {
    try {
      const results = await axios.get(SITES_URL, {
        headers: {
          authorization: `Bearer ${token}`,
        },
      })
      if (!results.data.content.ok) {
        throw new Error(results.data.content.error)
      }
      return results.data.content.sites
    } catch (error) {
      ErrorManager.captureError(error)
      throw error
    }
  },
  search: async (token = null, query) => {
    if (query && query.length < 3) {
      return {}
    }
    try {
      const results = await axios.get(ROOMS_URL, {
        headers: {
          authorization: `Bearer ${token}`,
        },
        params: {
          query,
        },
      })
      if (!results.data.content.ok) {
        throw new Error(results.data.content.error)
      }
      return results.data.content.rooms
    } catch (error) {
      ErrorManager.captureError(error)
      throw error
    }
  },
}

const workspacesApi = axios.create({
  baseURL: WORKSPACES_URL,
})

const workspacesController = {
  getLiveImage: async (token = null, { surveyId, mapId }) => {
    try {
      const result = await workspacesApi.get(`/getliveimage/map.svg`, {
        headers: {
          authorization: `Bearer ${token}`,
        },
        params: {
          map_id: mapId,
          survey_id: surveyId,
        },
        responseType: `arraybuffer`,
      })
      const image = btoa(
        new Uint8Array(result.data).reduce(
          (data, byte) => data + String.fromCharCode(byte),
          ``,
        ),
      )
      // there may be no need to base64 SVGs
      // but I have yet to test this extensively
      // https://css-tricks.com/probably-dont-base64-svg/
      return `data:image/svg+xml;base64,${image}`
    } catch (error) {
      ErrorManager.captureError(error)
      throw error
    }
  },
  getSummary: async (token = null) => {
    try {
      const { data, headers } = await workspacesApi.get(`/summary`, {
        headers: {
          authorization: `Bearer ${token}`,
        },
      })
      if (!data.content || data.error !== ``) {
        throw new Error(data.error || `There was a problem`)
      }
      return {
        data: data.content,
        lastModified: LocalisationManager.parseToMoment(
          headers[`last-modified`],
        ),
      }
    } catch (error) {
      ErrorManager.captureError(error)
      throw error
    }
  },
  getWorkspaces: async (token = null) => {
    try {
      const { data, headers } = await workspacesApi.get(`/`, {
        headers: {
          authorization: `Bearer ${token}`,
        },
      })
      if (!data.content || data.error !== ``) {
        throw new Error(data.error)
      }
      return {
        data: data.content,
        lastModified: LocalisationManager.parseToMoment(
          headers[`last-modified`],
        ),
      }
    } catch (error) {
      ErrorManager.captureError(error)
      throw error
    }
  },
}

const timetableApi = axios.create({
  baseURL: TIMETABLE_URL,
})

const timetableController = {
  getPersonalTimetable: async (
    token = null,
    date = LocalisationManager.getMoment(),
  ) => {
    const datePart = date
      ? `?date=${date.clone().startOf(`isoWeek`).format(`YYYY-MM-DD`)}`
      : ``
    const headers = {
      authorization: `Bearer ${token}`,
    }
    ErrorManager.addDetail({ headers })
    const results = await timetableApi.get(datePart, { headers })
    if (!results.data.content.ok || results.data.error !== ``) {
      throw new Error(
        results.data.error
        || `There was a problem fetching your personal timetable`,
      )
    }
    const { data: { content: { timetable } } } = results
    Object.keys(timetable).forEach((day) => {
      const dayTimetable = timetable[day]
      timetable[day] = {
        lastModified: LocalisationManager.parseToMoment(
          results.headers[`last-modified`],
        ),
        timetable: dayTimetable,
      }
    })
    return timetable
  },
  getPersonalWeekTimetable: async (
    token = null,
    date = LocalisationManager.getMoment().startOf(`isoWeek`),
  ) => {
    const queryParam = `?date=${date.format(`YYYY-MM-DD`)}`
    const headers = {
      authorization: `Bearer ${token}`,
    }
    ErrorManager.addDetail({ headers })
    const results = await timetableApi.get(`/week/${queryParam}`, {
      headers,
    })
    if (results.data.error !== ``) {
      throw new Error(
        results.data.error
        || `An error occurred when fetching your personal weekly timetable`,
      )
    }
    const { data: { content: { timetable } } } = results
    // eslint-disable-next-line sonarjs/no-identical-functions
    Object.keys(timetable).forEach((day) => {
      const dayTimetable = timetable[day]
      timetable[day] = {
        lastModified: LocalisationManager.parseToMoment(
          results.headers[`last-modified`],
        ),
        timetable: dayTimetable,
      }
    })
    return timetable
  },
}

export default {
  rooms: roomsController,
  timetable: timetableController,
  workspaces: workspacesController,
}
