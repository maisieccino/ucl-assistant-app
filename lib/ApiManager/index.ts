import PeopleController from './PeopleController'
import RoomsController from './RoomsController'
import TimetableController from './TimetableController'
import WorkspacesController from './WorkspacesController'

export default {
  people: PeopleController,
  rooms: RoomsController,
  timetable: TimetableController,
  workspaces: WorkspacesController,
}
