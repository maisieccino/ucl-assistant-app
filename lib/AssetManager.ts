import { Feather } from "@expo/vector-icons"

const uclapi = {
  iconForeground: require(`../assets/images/icon-fg.png`),
  smallIcon: require(`../assets/images/uclapi.png`),
  splash: require(`../assets/images/splash.png`),
}

const undraw = {
  buildingBlocks: require(`../assets/images/undraw_building_blocks.png`),
  calendar: require(`../assets/images/undraw_calendar.png`),
  chilling: require(`../assets/images/undraw_chilling.png`),
  dogWalking: require(`../assets/images/undraw_dog_walking.png`),
  graduation: require(`../assets/images/undraw_graduation.png`),
  notify: require(`../assets/images/undraw_notify.png`),
  peopleSearch: require(`../assets/images/undraw_people_search.png`),
  playfulCat: require(`../assets/images/undraw_playful_cat.png`),
  relaxation: require(`../assets/images/undraw_relaxation.png`),
  relaxingAtHome: require(`../assets/images/undraw_relaxing_at_home.png`),
  scrumBoard: require(`../assets/images/undraw_scrum_board.png`),
  studying: require(`../assets/images/undraw_studying.png`),
}

const font = {
  ...Feather.font,
  apercu: require(`../assets/fonts/somerandomfont.otf`),
  "apercu-bold": require(`../assets/fonts/somerandomfont-Bold.otf`),
  "apercu-light": require(`../assets/fonts/somerandomfont-Light.otf`),
}

export default {
  font,
  uclapi,
  undraw,
}
