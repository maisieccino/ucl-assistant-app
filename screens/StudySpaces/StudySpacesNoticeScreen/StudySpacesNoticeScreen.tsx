import React, { useCallback } from 'react'
import { View, StyleSheet, Image } from 'react-native'
import { Page } from '../../../components/Containers'
import { BodyText, SubtitleText, Link } from '../../../components/Typography'
import Button from "../../../components/Button"
import { AssetManager, WebBrowserManager } from '../../../lib'

const styles = StyleSheet.create({
  bookingButton: {
    marginTop: 20,
  },
  container: {},
  explanation: {
    marginBottom: 10,
    marginTop: 10,
  },
  image: {
    alignSelf: `center`,
    height: 200,
    marginTop: 5,
    width: 200,
  },
})

const StudySpacesNoticeScreen: React.FC = () => {
  const openBookingPage = useCallback(
    () => WebBrowserManager.openLink(`https://library-calendars.ucl.ac.uk/r/`),
    [],
  )

  return (
    <Page>
      <View style={styles.container}>
        <SubtitleText>StudySpaces</SubtitleText>
        <Image
          source={AssetManager.undraw.socialDistancing}
          resizeMethod="scale"
          resizeMode="contain"
          style={styles.image}
        />
        <BodyText style={styles.explanation}>
          Study spaces in libraries and in the various academic buildings are not
          available for walk-in use during the ongoing pandemic.
          They must be booked in advance.
        </BodyText>
        <BodyText splitByWord>
          You can find out more about the current policy around using the study spaces on the
          <Link splitByWord href="https://www.ucl.ac.uk/library/libraries-and-study-spaces/bookable-study-spaces-help">
            Library website
          </Link>
          .
        </BodyText>
        <Button onPress={openBookingPage} style={styles.bookingButton}>
          Book a StudySpace
        </Button>
      </View>
    </Page>
  )
}

export default StudySpacesNoticeScreen
