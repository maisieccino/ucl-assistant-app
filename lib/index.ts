import AnalyticsManager from './AnalyticsManager'
import ApiManager from './ApiManager'
import AssetManager from './AssetManager'
import debounce from './debounce'
import DeviceManager from './DeviceManager'
import ErrorManager from './ErrorManager'
import LocalisationManager from './LocalisationManager'
import MailManager from './MailManager'
import MapsManager from './MapsManager'
import PushNotificationsManager from './PushNotificationsManager'
import Shadow from './Shadow'
import WebBrowserManager from './WebBrowserManager'

const Random = {
  array: (arr) => arr[Math.floor(Math.random() * arr.length)],
}

export {
  AnalyticsManager,
  ApiManager,
  AssetManager,
  DeviceManager,
  ErrorManager,
  LocalisationManager,
  MailManager,
  MapsManager,
  PushNotificationsManager,
  Random,
  Shadow,
  WebBrowserManager,
  debounce,
}
