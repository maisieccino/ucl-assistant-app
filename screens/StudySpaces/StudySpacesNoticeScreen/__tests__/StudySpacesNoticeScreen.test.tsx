import { fireEvent } from "@testing-library/react-native"
import React from 'react'
import { render } from "../../../../jest/test-utils"
import { WebBrowserManager } from "../../../../lib"
import StudySpacesNoticeScreen from "../StudySpacesNoticeScreen"

const policyURL = `https://www.ucl.ac.uk/library/libraries-and-study-spaces/bookable-study-spaces-help`
const bookingURL = `https://library-calendars.ucl.ac.uk/r/`

describe(`StudySpacesNoticeScreen`, () => {
  const mockOpenLink = jest.fn(() => Promise.resolve())
  WebBrowserManager.openLink = mockOpenLink as any

  it(`renders the StudySpacesNoticeScreen`, async () => {
    const wrapper = await render(<StudySpacesNoticeScreen />)
    expect(wrapper).toMatchSnapshot()
  })

  it(`opens linked pages`, async () => {
    const wrapper = await render(<StudySpacesNoticeScreen />)
    const { getByText } = wrapper

    const policyLink = getByText(/Library/i)
    expect(policyLink.props.href === policyURL)
    fireEvent.press(policyLink)
    expect(mockOpenLink).toHaveBeenCalledTimes(1)
    expect(mockOpenLink).toHaveBeenCalledWith(policyURL)

    const bookingButton = getByText(/Book a StudySpace/i)
    fireEvent.press(bookingButton)
    expect(mockOpenLink).toHaveBeenCalledTimes(2)
    expect(mockOpenLink).toHaveBeenCalledWith(bookingURL)
  })
})
