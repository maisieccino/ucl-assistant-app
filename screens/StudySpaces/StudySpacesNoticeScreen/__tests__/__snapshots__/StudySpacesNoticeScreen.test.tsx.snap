// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StudySpacesNoticeScreen renders the StudySpacesNoticeScreen 1`] = `
Array [
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "backgroundColor": "#fefefe",
        "flex": 0,
      }
    }
  />,
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "#fefefe",
            "flex": 1,
            "flexDirection": "column",
          },
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <RCTScrollView
        contentContainerStyle={
          Array [
            Object {
              "paddingBottom": 60,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 20,
            },
            undefined,
          ]
        }
        keyboardDismissMode="on-drag"
        refreshControl={null}
        style={
          Array [
            Object {
              "backgroundColor": "#fefefe",
              "flex": 1,
              "paddingLeft": 0,
              "paddingRight": 0,
            },
            undefined,
          ]
        }
      >
        <View>
          <View
            style={Object {}}
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#4D5061",
                    "fontFamily": "apercu-bold",
                    "fontSize": 28,
                    "marginBottom": 2,
                    "marginTop": 8,
                  },
                  undefined,
                ]
              }
            >
              StudySpaces
            </Text>
            <Image
              resizeMethod="scale"
              resizeMode="contain"
              source={1}
              style={
                Object {
                  "alignSelf": "center",
                  "height": 200,
                  "marginTop": 5,
                  "width": 200,
                }
              }
            />
            <Text
              style={
                Array [
                  Object {
                    "color": "#4D5061",
                    "fontFamily": "apercu",
                    "fontSize": 16,
                    "marginBottom": 1,
                    "marginTop": 1,
                  },
                  Object {
                    "marginBottom": 10,
                    "marginTop": 10,
                  },
                ]
              }
            >
              Study spaces in libraries and in the various academic buildings are not available for walk-in use during the ongoing pandemic. They must be booked in advance.
            </Text>
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "flexWrap": "wrap",
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#4D5061",
                      "fontFamily": "apercu",
                      "fontSize": 16,
                      "marginBottom": 1,
                      "marginTop": 1,
                    },
                    undefined,
                  ]
                }
              >
                You
                 
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#4D5061",
                      "fontFamily": "apercu",
                      "fontSize": 16,
                      "marginBottom": 1,
                      "marginTop": 1,
                    },
                    undefined,
                  ]
                }
              >
                can
                 
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#4D5061",
                      "fontFamily": "apercu",
                      "fontSize": 16,
                      "marginBottom": 1,
                      "marginTop": 1,
                    },
                    undefined,
                  ]
                }
              >
                find
                 
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#4D5061",
                      "fontFamily": "apercu",
                      "fontSize": 16,
                      "marginBottom": 1,
                      "marginTop": 1,
                    },
                    undefined,
                  ]
                }
              >
                out
                 
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#4D5061",
                      "fontFamily": "apercu",
                      "fontSize": 16,
                      "marginBottom": 1,
                      "marginTop": 1,
                    },
                    undefined,
                  ]
                }
              >
                more
                 
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#4D5061",
                      "fontFamily": "apercu",
                      "fontSize": 16,
                      "marginBottom": 1,
                      "marginTop": 1,
                    },
                    undefined,
                  ]
                }
              >
                about
                 
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#4D5061",
                      "fontFamily": "apercu",
                      "fontSize": 16,
                      "marginBottom": 1,
                      "marginTop": 1,
                    },
                    undefined,
                  ]
                }
              >
                the
                 
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#4D5061",
                      "fontFamily": "apercu",
                      "fontSize": 16,
                      "marginBottom": 1,
                      "marginTop": 1,
                    },
                    undefined,
                  ]
                }
              >
                current
                 
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#4D5061",
                      "fontFamily": "apercu",
                      "fontSize": 16,
                      "marginBottom": 1,
                      "marginTop": 1,
                    },
                    undefined,
                  ]
                }
              >
                policy
                 
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#4D5061",
                      "fontFamily": "apercu",
                      "fontSize": 16,
                      "marginBottom": 1,
                      "marginTop": 1,
                    },
                    undefined,
                  ]
                }
              >
                around
                 
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#4D5061",
                      "fontFamily": "apercu",
                      "fontSize": 16,
                      "marginBottom": 1,
                      "marginTop": 1,
                    },
                    undefined,
                  ]
                }
              >
                using
                 
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#4D5061",
                      "fontFamily": "apercu",
                      "fontSize": 16,
                      "marginBottom": 1,
                      "marginTop": 1,
                    },
                    undefined,
                  ]
                }
              >
                the
                 
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#4D5061",
                      "fontFamily": "apercu",
                      "fontSize": 16,
                      "marginBottom": 1,
                      "marginTop": 1,
                    },
                    undefined,
                  ]
                }
              >
                study
                 
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#4D5061",
                      "fontFamily": "apercu",
                      "fontSize": 16,
                      "marginBottom": 1,
                      "marginTop": 1,
                    },
                    undefined,
                  ]
                }
              >
                spaces
                 
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#4D5061",
                      "fontFamily": "apercu",
                      "fontSize": 16,
                      "marginBottom": 1,
                      "marginTop": 1,
                    },
                    undefined,
                  ]
                }
              >
                on
                 
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#4D5061",
                      "fontFamily": "apercu",
                      "fontSize": 16,
                      "marginBottom": 1,
                      "marginTop": 1,
                    },
                    undefined,
                  ]
                }
              >
                the
                 
              </Text>
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#4D5061",
                        "fontFamily": "apercu",
                        "fontSize": 16,
                        "marginBottom": 1,
                        "marginTop": 1,
                      },
                      Object {
                        "color": "#2980b9",
                      },
                      Object {},
                    ]
                  }
                >
                  Library
                   
                </Text>
              </View>
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#4D5061",
                        "fontFamily": "apercu",
                        "fontSize": 16,
                        "marginBottom": 1,
                        "marginTop": 1,
                      },
                      Object {
                        "color": "#2980b9",
                      },
                      Object {},
                    ]
                  }
                >
                  website
                  
                </Text>
              </View>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#4D5061",
                      "fontFamily": "apercu",
                      "fontSize": 16,
                      "marginBottom": 1,
                      "marginTop": 1,
                    },
                    undefined,
                  ]
                }
              >
                .
                 
              </Text>
            </View>
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "opacity": 1,
                }
              }
            >
              <ViewManagerAdapter_ExpoLinearGradient
                proxiedProperties={
                  Object {
                    "colors": Array [
                      4279998859,
                      4283151357,
                    ],
                    "endPoint": Array [
                      1,
                      0,
                    ],
                    "locations": undefined,
                    "startPoint": Array [
                      0,
                      1,
                    ],
                  }
                }
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "borderRadius": 8,
                      "elevation": 2,
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "marginBottom": 5,
                      "marginTop": 5,
                      "minHeight": 55,
                      "minWidth": 150,
                      "padding": 8,
                      "paddingLeft": 16,
                      "paddingRight": 16,
                      "shadowColor": "#1B998B",
                      "shadowOffset": Object {
                        "height": 2,
                        "width": 0,
                      },
                      "shadowOpacity": 0.75,
                      "shadowRadius": 3,
                    },
                    Object {
                      "marginTop": 20,
                    },
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#fefefe",
                        "fontFamily": "apercu",
                        "fontSize": 20,
                        "marginBottom": 1,
                        "marginTop": 1,
                        "textAlign": "center",
                      },
                      undefined,
                    ]
                  }
                >
                  Book a StudySpace
                </Text>
              </ViewManagerAdapter_ExpoLinearGradient>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </RCTSafeAreaView>,
]
`;
